<launch>
<!-- Launch file for Livox AVIA LiDAR -->

	<arg name="rviz" default="true" />
	<arg name="use_sim_time" value="true"/>
	<arg name="map" default="$(find fast_lio)/PCD/rainbow.pcd" />

	<!-- fast_lio -->
	<rosparam command="load" file="$(find fast_lio_localization)/config/mid360.yaml" />
	<param name="feature_extract_enable" type="bool" value="0"/>
	<param name="point_filter_num" type="int" value="3"/>
	<param name="max_iteration" type="int" value="3" />
	<param name="filter_size_surf" type="double" value="0.5" />
	<param name="filter_size_map" type="double" value="0.5" />
	<param name="cube_side_length" type="double" value="1000" />
	<param name="runtime_pos_log_enable" type="bool" value="false" />
    <param name= "pcd_save_en" type="bool" value="false"/>
    <node pkg="fast_lio" type="fastlio_mapping" name="laserMapping" output="screen" /> 

	<!-- loalization-->
    <node pkg="fast_lio_localization" type="global_localization.py" name="global_localization" output="screen" />

	<!-- transform  fusion-->
    <node pkg="fast_lio_localization" type="transform_fusion.py" name="transform_fusion" output="screen" />

	<!-- glbal map-->
    <node pkg="pcl_ros" type="pcd_to_pointcloud" name="map_publishe" output="screen"
		  args="$(arg map) 5 _frame_id:=map cloud_pcd:=map" />

	<group if="$(arg rviz)">
		<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio_localization)/rviz_cfg/localization.rviz" />
	</group>

	<!-- load 2d map -->
	<arg name="2dmap" default="scans.yaml" />
	<node name = "map_server" pkg = "map_server" type = "map_server" args="$(find fast_lio)/PCD/scans.yaml /map:=prior_map"/>

	<!-- pointscloud2 to laserscans -->
	<include file="$(find fast_lio_localization)/launch/PointsCloud2toLaserscan.launch" />
</launch>
